### ISSUE TITLE
SQL error when configuring Constant Contact API connection (accessToken is too long)

### ISSUE THREAD

**Issue by wrux** [solspace/craft3-freeform#590]
**Describe the bug or issue you're experiencing**

I'm getting an SQL error when trying to setup the Constant Contact API connection.

When clicking on the "Click here to re-authorize" button and go through the setup, it sends me back to Craft and shows a 500 error. 

Here's the error and it looks like the access token is 1000+ chars long. 

Error:

```
Next yii\db\Exception: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'accessToken' at row 1
The SQL being executed was: UPDATE `craft_freeform_integrations` SET `accessToken`='', `settings`='{\"return_uri\":\"{{url}}/admin/freeform/settings/mailing-lists/constantContact\",\"client_id\":\"\",\"client_secret\":\"\",\"refresh_token\":\"\"}', `lastUpdate`='2023-02-06 03:21:44', `dateUpdated`='2023-02-06 03:21:44' WHERE `id`=1 in /vendor/yiisoft/yii2/db/Schema.php:678
```

Looks like `accessToken` is `varchar(255)` but should be much longer

```sql
+-------------+--------------+------+-----+---------+----------------+
| Field       | Type         | Null | Key | Default | Extra          |
+-------------+--------------+------+-----+---------+----------------+
| id          | int(11)      | NO   | PRI | NULL    | auto_increment |
| name        | varchar(255) | NO   |     | NULL    |                |
| handle      | varchar(255) | NO   | UNI | NULL    |                |
| type        | varchar(50)  | NO   | MUL | NULL    |                |
| class       | varchar(255) | YES  |     | NULL    |                |
| accessToken | varchar(255) | YES  |     | NULL    |                |
| settings    | text         | YES  |     | NULL    |                |
| forceUpdate | tinyint(1)   | YES  |     | 0       |                |
| lastUpdate  | datetime     | YES  |     | NULL    |                |
| dateCreated | datetime     | NO   |     | NULL    |                |
| dateUpdated | datetime     | NO   |     | NULL    |                |
| uid         | char(36)     | NO   |     | 0       |                |
+-------------+--------------+------+-----+---------+----------------+
```

**Steps to reproduce**

1. Create new Constant Contact connection
2. Setup authentication
3. Authorise app 
4. See 500 error on return to Craft

**Expected behavior**

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.6.4
- Freeform Version: 3.13.22.1
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Additional context**

**Comment by kjmartens** [comment]
Sorry for the trouble @wrux.

Just to confirm, did you copy and paste the **Redirect URI** value inside Freeform settings to the application inside Constant Contact? Also, would you be able to delete that Constant Contact integration (and maybe even the application you created in Constant Contact) and try creating it all over again and see if it makes any difference? 🙂

**Comment by wrux** [comment]
@kjmartens Yes I have setup the connection exactly as described [here](https://docs.solspace.com/craft/freeform/v3/integrations/email-marketing/constant-contact.html).

I just deleted the integration in Craft, ran through the setup again with the same result. After clicking on re-authorise, it redirects to Constant Contact, then back to Craft on successful authorisation and I get an SQL error. The token it's trying to save in the DB is like 1000+ chars long, but the database field for `accessToken` is `varchar(255)` so too small. 

After going into the database and changing `accessToken` to TEXT, the connection works just fine and form submissions are sending to Constant Contact. 

So should the datatype be changed in the plugin?

Here's the SQL error (with sensitive data removed):

```sql
Next yii\db\Exception: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'accessToken' at row 1
The SQL being executed was: 

UPDATE `craft_freeform_integrations` 
SET `accessToken`='', `settings`='{\"return_uri\":\"/admin/freeform/settings/mailing-lists/constantContact\",\"client_id\":\"\",\"client_secret\":\"\",\"refresh_token\":\"-\"}', `lastUpdate`='2023-02-08 05:06:30', `dateUpdated`='2023-02-08 05:06:30' 
WHERE `id`=3 

in /vendor/yiisoft/yii2/db/Schema.php:678
```

**Comment by kjmartens** [comment]
I will have a developer check into this and get back to you. 🙂

**Comment by kjmartens** [comment]
Hi again @wrux,

I just checked my install and mine is set to `TEXT` already. There was a migration to address this for existing sites, but it looks like the fresh install script wasn't updated. That is fixed in the repo now, and will be available in 3.13.24 (Freeform 4 already has this fix).

You don't need to take any further action. The manual change you made was the correct one. 🙂
